packages/components/src/RichTooltip/RichTooltip.component.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Actions } from '../Actions';
+import CircularProgress from '../CircularProgress';
+import Icon from '../Icon';
+import Inject from '../Inject';
+import theme from './RichTooltip.scss';
+
+export function getContent(props) {
+	let content;
+
+	if (typeof props.Content === 'string') {
+		content = props.Content;
+	}
+
+	if (props.loading) {
+		content = (
+			<div className={theme['tc-center-content']}>
+				<CircularProgress />
+			</div>
+		);
+	}
+
+	if (props.error) {
+		content = (
+			<div className={classNames(theme['tc-center-content'], theme['tc-error-wrapper'])}>
+				<Icon name="talend-warning" className={theme['tc-center-icon']} />
+				<div className={theme['tc-error-body']}>
+					<h1>Whoops !</h1>
+					<div>{props.error}</div>
+				</div>
+			</div>
+		);
+	}
+
+	return content;
+}
+
+export function RichTooltipContent(props) {
+	const content = getContent(props);
+
+	if (content) {
+		return <RichTooltip.Body className={props.className} Content={content} />;
+	}
+
+	return Inject.getReactElement(props.getComponent, props.Content);
+}
+
+RichTooltipContent.propTypes = {
+	className: PropTypes.string,
+	getComponent: PropTypes.func,
+	loading: PropTypes.bool,
+	Content: Inject.getReactElement.propTypes,
+	error: PropTypes.string,
+};
+
+export function RichTooltipBody(props) {
+	// fix ie11 flexbox align-items: center with min-height https://stackoverflow.com/a/47180276
+	return (
+		<div
+			className={classNames(props.className, {
+				[theme['tc-tooltip-body']]: true,
+				'tc-tooltip-body': true,
```

Comment:
if you don t have condition on it you can just use the args syntax instead of object.